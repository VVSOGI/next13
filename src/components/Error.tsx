'use client'
import React from 'react'

interface Props {
    message: string
}

export function Error({ message }: Props) {
    return (
        <div>
            <h2>{message}</h2>
            <button>Try again</button>
        </div>
    )
}
