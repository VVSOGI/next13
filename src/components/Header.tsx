import React from 'react'

export function Header() {
    return (
        <div
            style={{
                padding: '3rem 3rem',
                borderBottom: '1px solid #eaeaea',
                flex: 'none'
            }}
        >
            Header
        </div>
    )
}
