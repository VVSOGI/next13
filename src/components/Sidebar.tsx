import React from 'react'

interface Props {
    style?: React.CSSProperties
}

export function Sidebar({ style }: Props) {
    return (
        <div
            style={{
                ...style,
                display: 'flex',
                justifyContent: 'center',
                alignItems: 'center',
                height: '100%',
                borderRight: '1px solid #eaeaea'
            }}
        >
            Sidebar
        </div>
    )
}
