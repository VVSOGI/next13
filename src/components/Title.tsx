import React, { useState } from 'react'

interface Props {
    title: string
    changeTitle: (title: string) => void
}

export function Title({ title, changeTitle }: Props) {
    return (
        <div style={{ display: 'flex', flexDirection: 'column' }}>
            {title}
            <input
                style={{ width: '400px', padding: '1rem', marginTop: '1rem' }}
                placeholder="Enter your home title"
                type="text"
                onChange={(e) => changeTitle(e.target.value)}
            />
        </div>
    )
}
