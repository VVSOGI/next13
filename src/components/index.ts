'use client'

export * from './Header'
export * from './Title'
export * from './projects'
export * from './Error'
