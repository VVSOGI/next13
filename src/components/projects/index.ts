export * from './ProjectsContainer'
export * from './ProjectContents'
