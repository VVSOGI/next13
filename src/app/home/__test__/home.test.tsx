import { fireEvent, render, renderHook } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Title } from '@/components'
import useModel from '../model'

describe('Home', () => {
    const hooks = renderHook(() => useModel(), {})
    const {
        result: {
            current: { changeTitle }
        }
    } = hooks

    test('renders a No Title Props', () => {
        const screen = render(<Title title="Test Title1" changeTitle={changeTitle} />)

        expect(screen.container).toMatchSnapshot()
    })

    test('renders a Title Props', () => {
        const screen = render(<Title title="Test Title2" changeTitle={changeTitle} />)

        const input = screen.getByPlaceholderText('Enter your home title')
        fireEvent.change(input, { target: { value: 'Test Title1' } })

        expect(screen.container).toMatchSnapshot()
    })
})
