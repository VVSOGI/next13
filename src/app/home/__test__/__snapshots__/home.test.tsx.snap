// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a No Title Props 1`] = `
<div>
  <div
    style="display: flex; flex-direction: column;"
  >
    Test Title1
    <input
      placeholder="Enter your home title"
      style="width: 400px; padding: 1rem; margin-top: 1rem;"
      type="text"
    />
  </div>
</div>
`;

exports[`Home renders a Title Props 1`] = `
<div>
  <div
    style="display: flex; flex-direction: column;"
  >
    Test Title2
    <input
      placeholder="Enter your home title"
      style="width: 400px; padding: 1rem; margin-top: 1rem;"
      type="text"
    />
  </div>
</div>
`;
